### ISSUE TITLE
Occasionally changed configMap is not deployed immediately

### ISSUE THREAD

**Issue by cocowowo** [spinnaker/spinnaker#4795]
### Issue Summary:
I set up the Deploy (manifest) stage in my pipeline.
Sometimes when I modify configmap and run pipeline, the changed configMap is not deployed immediately.
After a while, when I run the pipeline again, the changed configMap is deployed.

### Cloud Provider(s):
aws

### Environment:
I am running Spinnaker to deploy pods into AWS EKS.

### Feature Area (if this issue is UI/UX related, please tag `@spinnaker/ui-ux-team`):
orca

### Steps to Reproduce:
1. Configure pipeline. 
2. Edit some configMap of Deploy(manifest) stage
3. Click Save button and wait until you see "In sync with server"
4. Run pipeline and check execution detail.

**Comment by maggieneterval** [comment]
@cocowowo thanks for reporting this issue! Would you mind providing your pipeline JSON and ConfigMap YAML (with any sensitive info removed, of course) so we can try to reproduce? Thanks!

**Comment by cocowowo** [comment]
@maggieneterval 
Here is my pipeline JSON.
```
{
  "keepWaitingPipelines": false,
  "lastModifiedBy": "cocowowo",
  "limitConcurrent": true,
  "parameterConfig": [],
  "stages": [
    {
      "account": "my-kube-account",
      "cloudProvider": "kubernetes",
      "manifestArtifactAccount": "docker-registry",
      "manifests": [
        {
          "apiVersion": "v1",
          "data": {
            "DB_PASS": "mypass5",
            "DB_URL": "localhost6",
            "DB_USER": "myuser7",
            "DEBUG_INFO": "debug9"
          },
          "kind": "ConfigMap",
          "metadata": {
            "name": "config-dev",
            "namespace": "spinnaker"
          }
        }
      ],
      "moniker": {
        "app": "demo"
      },
      "name": "Deploy (Manifest)",
      "refId": "1",
      "relationships": {
        "loadBalancers": [],
        "securityGroups": []
      },
      "requisiteStageRefIds": [],
      "skipExpressionEvaluation": false,
      "source": "text",
      "trafficManagement": {
        "enabled": false,
        "options": {
          "enableTraffic": false,
          "services": []
        }
      },
      "type": "deployManifest"
    }
  ],
  "triggers": [],
  "updateTs": "1566799719000"
}
```
I changed DB_PASS or DB_URL and saved pipeline.
And then if I run the pipeline right away the problem is sometimes reproduced.

**Comment by cocowowo** [comment]
> > And then if I run the pipeline right away the problem is sometimes reproduced.
> 
> Curious if you can confirm that the pipeline saved successfully prior to the execution -- if you inspect the execution JSON of an execution where it did not deploy successfully (by clicking "Source" under the pipeline execution details), can you confirm that the manifest contains the DB_PASS or DB_URL value you just saved?

I inspect execution JSON by clicking "Source" when problem reproduced.  The DB_PASS value  does 
 not changed even though I confirmed it was saved(showing "In sync with server").
Is it normal? How long does it take to save safely?

**Comment by maggieneterval** [comment]
Thanks for confirming, sounds like the root of the issue is that the UI is telling you your pipeline has saved successfully, but an immediate manual execution somehow is still pointing to the pre-saved version. I'm not able to reproduce this on my end (but maybe I'm not as fast at saving + immediately executing as you are!) but definitely keep me posted if you see this happening a lot, and if possible keep your browser console open to check for any error message or failed network requests that could help us get to the bottom of this!

**Comment by ezimanyi** [comment]
This may be the issue that was fixed in 1.14 by spinnaker/echo#518.
